src/shogun/preprocessor/DensePreprocessor.h

```diff
@@ -76,6 +78,10 @@ template <class ST> class DensePreprocessor : public Preprocessor
 		 * @return the matrix after applying the preprocessor
 		 */
 		virtual SGMatrix<ST> inverse_apply_to_matrix(SGMatrix<ST> matrix);
+
+		virtual void fit_impl(const SGMatrix<ST>& feature_matrix)
```

Comment:
no error in here?

Reply:
I like the SGMarix!

Reply:
what about inplace? it is const

Reply:
The idea is that you shouldn’t get an error if the Transformer doesn’t implement a fit method because it doesn’t make sense. But I think this should be pure virtual function and then each implementation handles this.. that way nothing unexpected happens..

Reply:
could you explain why it doesnt make sense? 
I am ok with a purely virtual method, just needs more refactoring

Any thoughts about the const matrix and inplace?

Reply:
> could you explain why it doesnt make sense?

ah sorry, yes. It doesn't make sense for something like `LogPlusOne`. There is nothing to `fit`, only to `apply`. So if I put this `Transformer` in a `Pipeline` it will always throw.

> Any thoughts about the const matrix and inplace?

Isn't the inplace for `transform` rather than `fit`?

Reply:
ah yes, sorry I keep confusing the two.
See above comment on error/purely-virtual/no-op